netket/nn/utils.py

```diff
@@ -90,6 +91,11 @@ def _to_array_rank(apply_fun, variables, σ_rank, n_states, normalize, allgather
     Args:
         n_states: total number of elements in the hilbert space.
     """
+    
+    if chunk_size is not None:
+        apply_fun = nkjax.apply_chunked(apply_fun, in_axes=(None, 0), chunk_size=chunk_size)
```

Comment:
apply_chunked already returns the original function if chunk_size is None...

```suggestion
    apply_fun = nkjax.apply_chunked(apply_fun, in_axes=(None, 0), chunk_size=chunk_size)
```